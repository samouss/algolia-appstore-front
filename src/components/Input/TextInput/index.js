import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './index.css';

const TextInput = ({ value, onChange, enableClearButton, ...props }) => (
  <div styleName="TextInput">
    <input
      {...props}
      styleName={cx('TextInput__Input', {
        'TextInput__Input--clear': enableClearButton,
      })}
      value={value}
      onChange={onChange}
    />
    {enableClearButton && value && (
      <i
        styleName="TextInput__Clear"
        className="fa fa-times-circle"
        aria-hidden="true"
        onClick={() => onChange({
          currentTarget: {
            value: '',
          },
        })}
      />
    )}
  </div>
);

TextInput.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  enableClearButton: PropTypes.bool,
};

TextInput.defaultProps = {
  enableClearButton: true,
};

export default TextInput;
