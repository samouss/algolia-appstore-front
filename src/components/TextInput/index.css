.TextInput {
  -webkit-appearance: none;
  border: 1px solid #f2f2f2;
  padding: 10px;
  font-family: "Helvetica Neue Light", "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
  font-size: 1.3rem;
  font-weight: 300;

  &:focus {
    outline: none;
  }
  &::placeholder {
    opacity: 1;
    color: #b9b9b9;
  }
}
