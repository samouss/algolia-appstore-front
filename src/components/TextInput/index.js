import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const TextInput = ({ value, onChange, ...props }) => (
  <input
    {...props}
    styleName="TextInput"
    value={value}
    onChange={onChange}
  />
);

TextInput.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default TextInput;
