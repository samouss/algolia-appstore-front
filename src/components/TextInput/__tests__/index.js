import React from 'react';
import { shallow } from 'enzyme';
import TextInput from '../index';

describe('<TextInput />', () => {
  const defaultProps = {
    value: 'http://link.com',
    onChange: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
      placeholder: 'Hello',
    };

    const component = shallow(
      <TextInput
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onChange', () => {
    const props = {
      ...defaultProps,
      onChange: jest.fn(),
    };

    const component = shallow(
      <TextInput
        {...props}
      />,
    );

    component
      .find('input')
      .simulate('change');

    expect(props.onChange).toHaveBeenCalled();
  });
});
