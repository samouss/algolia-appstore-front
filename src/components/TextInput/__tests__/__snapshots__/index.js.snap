// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> expect to render 1`] = `
<input
  onChange={[Function]}
  placeholder="Hello"
  styleName="TextInput"
  value="http://link.com"
/>
`;
